providers/oci/oci.go

```diff
@@ -295,19 +297,31 @@ func NewBucket(logger log.Logger, ociConfig []byte) (*Bucket, error) {
 	provider := Provider(strings.ToLower(config.Provider))
 	level.Info(logger).Log("msg", "creating OCI client", "provider", provider)
 	switch provider {
-	case DefaultConfigProvider:
+	case defaultConfigProvider:
 		configurationProvider = common.DefaultConfigProvider()
-	case InstancePrincipalConfigProvider:
+	case instancePrincipalConfigProvider:
 		configurationProvider, err = auth.InstancePrincipalConfigurationProvider()
 		if err != nil {
 			return nil, errors.Wrapf(err, "unable to create OCI instance principal config provider")
 		}
-	case RawConfigProvider:
+	case rawConfigProvider:
 		if err := config.validateConfig(); err != nil {
 			return nil, errors.Wrapf(err, "invalid oci configurations")
 		}
 		configurationProvider = common.NewRawConfigurationProvider(config.Tenancy, config.User, config.Region,
 			config.Fingerprint, config.PrivateKey, &config.Passphrase)
+	case okeWorkloadIdentityConfigProvider:
+		if err := os.Setenv(auth.ResourcePrincipalVersionEnvVar, auth.ResourcePrincipalVersion2_2); err != nil {
```

Comment:
Even though it makes me feel weird to set environment variables to configure the configuration provider, as far as I can understand from glancing at the SDK code, this is the only way :/

Reply:
https://github.com/oracle/oci-go-sdk/blob/604c12bf63a5efd6e63a073a384bf4f86fcec150/common/auth/resouce_principal_key_provider.go#L99

Reply:
Agreed